kubernetes/go-workspace-pull/README.md

```diff
@@ -0,0 +1,184 @@
+                         | e378c0810e9eac12e16                           |           |
+  pull_tag               | dev                                           | string    | default
+  push_registry_password | 122bc444b708853582dafe3cda069461b2a2c6ebce9a9 | sensitive | file
+                         | d38aee531a98d6d3d03                           |           |
+  push_registry_username | 890e8f907513987332a9679089def709997ac359a11e8 | sensitive | file
+                         | e378c0810e9eac12e16                           |           |
+  push_tag               | dev                                           | string    | default
+  port                   |                                          3000 | int       | default
+  pull_image             | catsby/example-go-pull                        | string    | default
+  push_image             | catsby/example-go-pull                        | string    | file
+```
+
+This will create the first docker image and push it up to the registry defined in the `dev.wpvars` file.
+
+Specifically note that:
+
+ - The output from Kaniko demonstrates actually building the container image
+ - the `SOURCE` for the `push_` variables are read from file
+ - the `SOURCE` for the `pull_` variables use the defaults because they were not overriden in the `dev.wpvars` file
+
+### Example: Creating a production build
+
+Users can then use the dev image with the `docker-pull` plugin by using another
+wpvars file, `prod.wpvars`, which adds the `pull_` variables and changes the
+`push_` variables. In this example, the values use for `push_` variables in
+`dev.wpvars` are now the values used for the `pull_` variable values, and new
+`push_` values are added in the `prod.wpvars` file. Technically they could be
+the same values for both, as the `push_tag` and `pull_tag` are not overridden in
+either wpvar files and are coded in the `waypoint.hcl` file. This was done to
+ensure default workspace builds push with the `dev` tag and production workspace
+builds pull with the `dev` tag and push with the `latest` tag.
+
+```
+# prod.wpvars 
+// pull from the dev registry
+pull_image="dev-registry-url/example-go-pull"
+pull_registry_username="developer"
+pull_registry_password="some password"
+
+// push to the prod registry
+push_image="prov-registry-url/example-go-pull"
+push_registry_username="releaser"
+push_registry_password="some password"
+```
+
+Run a production build with `waypoint build`, referencing the prod wpvars file:
+
+```
+$ waypoint build -workspace=production -var-file=prod.wpvars
+
+» Building go-k8s...
+  Performing operation on "kubernetes" with runner profile "01GGACDWSQZ1BR3TNR8EW0KY1A"
+
+» Cloning data from Git
+  URL: https://github.com/hashicorp/waypoint-examples.git
+  Ref: go-workspace-pull
+
+» Downloading from Git
+  Git Commit: a59c6a6a518bc22259db35cc0ba02b1dd56d4483
+   Timestamp: 2022-10-28 17:19:00 +0000 UTC
+     Message: more updates
```

Comment:
```suggestion
» Cloning data from Git
         URL: https://github.com/hashicorp/waypoint-examples.git
         Ref: go-workspace-pull
  Git Commit: a59c6a6a518bc22259db35cc0ba02b1dd56d4483
   Timestamp: 2022-10-28 17:19:00 +0000 UTC
     Message: more updates
```

This UX was updated recently, so we might as well get it updated here too!

Reply:
Excellent, thanks!